src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1694,8 +1742,16 @@ PlanRouterQuery(Query *originalQuery, RelationRestrictionContext *restrictionCon
 
 		Assert(UpdateOrDeleteQuery(originalQuery));
 
+		if (list_length(
+				plannerRestrictionContext->relationRestrictionContext->
+				relationRestrictionList) > 1)
+		{
+			isMultipleTableModifyQuery = true;
+		}
+
 		planningError = ModifyQuerySupported(originalQuery, originalQuery,
-											 isMultiShardQuery);
+											 isMultipleTableModifyQuery,
```

Comment:
As discussed on slack, we're passing the wrong variable here. `isMultipleTableModifyQuery` should be `isMultiShardQuery` and we don't need `isMultipleTableModifyQuery` here at all